docs/src/test/java/sample/javadsl/RunnableGraphStreamWrapperActor.java

```diff
@@ -0,0 +1,194 @@
+            Outlet<EventMessage> unknownEventOutlet = eventTypeFanOut.out(0);
+            Outlet<EventMessage> firstPartitionOutlet = eventTypeFanOut.out(1);
+            Outlet<EventMessage> secondPartitionOutlet = eventTypeFanOut.out(2);
+
+            builder.from(sourceShape)
+                    .toFanOut(eventTypeFanOut)
+                        .from(unknownEventOutlet)
+                            .to(builder.add(commitOnlySink))
+                        .from(firstPartitionOutlet)
+                            .via(builder.add(firstPartitionFlow))
+                            .via(builder.add(produceMessageFlow))
+                            .to(builder.add((producerCommitSink)))
+                        .from(secondPartitionOutlet)
+                            .via(builder.add(secondPartitionFlow))
+                            .to(builder.add(commitOnlySink));
+
+            return ClosedShape.getInstance();
+        });
+
+        //#errorHandlingClosedRunnableGraph
+        // Self kill this actor so that BackoffSupervisor starts this actor(pipeline/stream) again!
+        RunnableGraph.fromGraph(completionStageGraph)
+                .run(materializer)
+                .isShutdown()
+                .whenComplete((done, throwable) -> {
+                    getSelf().tell(PoisonPill.getInstance(), getSelf());
+                });
+        //#errorHandlingClosedRunnableGraph
+
+    }
+
+    private EventMessage processMessage(EventMessage tuple2) {
+        // main business logic to process message
+        return tuple2;
+    }
+
+    private ProducerMessage.Message<String, String, ConsumerMessage.Committable> produceCommittableEvent(EventMessage eventTuple, String topic) {
+        return new ProducerMessage.Message<>(new ProducerRecord<String, String>(topic, "Produce New Message"), eventTuple.committableMessage.committableOffset());
+    }
+
+    /**
+     * Custom decider to decide the Supervision strategy.
+     */
+    private Function<Throwable, Supervision.Directive> stopOnSpecificExceptionDecider(ActorRef actorRef) {
+        return exception -> {
+
+            if (exception instanceof IllegalStateException) { // Any specific exception
+                return Supervision.stop(); // BackoffSupervisor will start again!
+            }
+            return Supervision.resume();
+        };
+    }
+
+    // Call it from application bootstrap when you want to kick off the pipeline
+    public static final void createSupervisor(ActorSystem system) {
+        Props childProps = Props.create(RunnableGraphStreamWrapperActor.class);
+
+        final Props supervisorProps = BackoffSupervisor.props(
+                Backoff.onStop(
+                        childProps,
+                        "streamActor",
+                        Duration.create(3, TimeUnit.SECONDS),
+                        Duration.create(30, TimeUnit.SECONDS),
+                        0.2));
+
+        system.actorOf(supervisorProps, "streamActorSupervisor");
```

Comment:
it's  better if this is only returning `Props` and then the actual creation of the actor is up to someone else, e.g. could be used as a child of another actor, not only as a top level actor.